### PR TITLE
feat(G2G): add presence

### PR DESCRIPTION
**Pull request by Mattis44**
## Description 
Add a presence for G2G marketplace.
I added the discord presence support for the popular gamers' marketplace named G2G.

## Acknowledgements
- [☑️ ] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [☑️ ] I linted the code by running `yarn format`
- [☑️] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots

Presence changing from user choice.
Selected categories from user.
Viewing a profile.
 
image
image
image
image

### CODE REVIEW

**Comment by EncryptedDev** [comment]
Low quality pull request. Refactor the codebase before reopening the pull request

**Comment by Mattis44** [comment]
> Low quality pull request. Refactor the codebase before reopening the pull request

What was wrong ?

File: websites/G/G2G/metadata.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  "author": {
+    "id": "283757164923715587",
+    "name": "Mattis.#0001"
+  },
+  "description": {
+    "en": "G2G is a world-leading digital marketplace, founded by serious technopreneurs."
+  },
+  "url": "www.g2g.com",
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/XkzTOsp.png",
```
Comment (DarkVillager) [inline websites/G/G2G/metadata.json:13]: Logo's should be 512x512

File: websites/G/G2G/presence.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
+const presence = new Presence({
+		clientId: "1073167802296438875",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/XkzTOsp.png",
```
Comment (DarkVillager) [inline websites/G/G2G/presence.ts:8]: Same for this one. It should be 512x512.

**Comment by Mattis44** [comment]
Ok, i will reopen a pr without theses changes. Thanks !

File: websites/G/G2G/presence.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+		startTimestamp: browsingTimestamp,
+	};
```
Comment (DarkVillager) [inline websites/G/G2G/presence.ts:10]: let's first create these variables so we can use the through out the code
```suggestion
	},
	{pathname, href} = document.location,
	path = pathname.split("/");
```

File: websites/G/G2G/presence.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+	switch (window.location.hostname) {
+		case "www.g2g.com": {
+			const path = document.location.pathname.split("/");
```
Comment (DarkVillager) [inline websites/G/G2G/presence.ts:14]:
```suggestion
```

File: websites/G/G2G/presence.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/XkzTOsp.png",
+		startTimestamp: browsingTimestamp,
+	};
+
+	switch (window.location.hostname) {
+		case "www.g2g.com": {
```
Comment (DarkVillager) [inline websites/G/G2G/presence.ts:13]: There are no other domains so remove this.

File: websites/G/G2G/presence.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+					document.querySelector<HTMLBodyElement>("div.text-h4").textContent;
+			} else if (document.location.pathname.includes("secure")) {
+				const item: string =
+					document.querySelector<HTMLBodyElement>("div.col-8.row").textContent;
+				presenceData.details = `Buying ${item}`;
+				presenceData.startTimestamp = browsingTimestamp;
+			} else if (path[1] !== "") {
+				presenceData.details = `Viewing ${path[1]}'s profile`;
+				presenceData.startTimestamp = browsingTimestamp;
+			}
```
Comment (DarkVillager) [inline websites/G/G2G/presence.ts:59]: Change this to a switch statement. Also add buttons to most pages (not to the homepage, search etc.) 

E.g.
```
switch (path[1]) {
	case "categories": {
		presenceData.details = "Browsing a category";
		presenceData.state =
			document.querySelector<HTMLHeadingElement>("title").textContent;
		presenceData.buttons = [
			{
				label: "Browse Category",
				url: href,
			},
		];
		break;
	}
}
```

**Review by DarkVillager — COMMENTED** [review]
In general if something is set at the top like startTimestamp you don't have to set it again at every check.

File: websites/G/G2G/metadata.json [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  "thumbnail": "https://i.imgur.com/4JlsyFM.png",
+  "color": "#f03827",
+  "tags": [
+    "marketplace",
+    "deals",
+    "games"
+  ],
+  "category": "games"
+}
```
Comment (DarkVillager) [inline websites/G/G2G/metadata.json:22]: Fix format & logo size
```suggestion
{
	"$schema": "https://schemas.premid.app/metadata/1.7",
	"author": {
		"name": "Mattis.",
		"id": "283757164923715587"
	},
	"service": "G2G",
	"description": {
		"en": "G2G is a world-leading digital marketplace, founded by serious technopreneurs."
	},
	"url": "www.g2g.com",
	"version": "1.0.0",
	"logo": "https://i.imgur.com/U0FQhL3.png",
	"thumbnail": "https://i.imgur.com/4JlsyFM.png",
	"color": "#f03827",
	"category": "games",
	"tags": ["marketplace", "deals", "games"],
	"settings": [
		{
			"id": "privacy",
			"title": "Privacy Mode",
			"icon": "fas fa-user-secret",
			"value": false
		},
		{
			"id": "buttons",
			"title": "Show Buttons",
			"icon": "fas fa-compress-arrows-alt",
			"value": true,
			"if": {
				"privacy": false
			}
		}
	]
}

```

File: websites/G/G2G/presence.ts (hunk 2 above)
Comment (DarkVillager) [inline websites/G/G2G/presence.ts:8]:
```suggestion
		largeImageKey: "https://i.imgur.com/U0FQhL3.png",
```

File: websites/G/G2G/presence.ts [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+			break;
+		}
+	}
+
```
Comment (DarkVillager) [inline websites/G/G2G/presence.ts:63]:
```suggestion
	if (!buttons || privacy && presenceData.buttons) delete presenceData.buttons;
	if (privacy && presenceData.state) delete presenceData.state;
```

**Comment by Mattis44** [comment]
> In general if something is set at the top like startTimestamp you don't have to set it again at every check.

Hello ! thanks for your messages ! 

I reopened a PR so i will take your changes.

**Pull request closed without merging**